### PR TITLE
Rename chunksize-tolerance option

### PR DESCRIPTION
**Pull request by phofl**
- [ ] Closes #xxxx
- [ ] Tests added / passed
- [ ] Passes `pre-commit run --all-files`

@dcherian rightfully pointed out that this option is applicable in other areas as well, so removing the shuffle part. I think we should be able to get away with this since it's only been one release?

### CODE REVIEW

**Review by jrbourbeau — COMMENTED** [review]
> I think we should be able to get away with this since it's only been one release?

Probably. Can you remind me what has happened here with this config name?

**Comment by phofl** [comment]
We are doing our best to preserve input chunk sizes, sometimes we can't and this option configures how much we can deviate from the input chunk size upwards.

**Comment by jrbourbeau** [comment]
Sorry, I meant what happened with the config option name itself. It looks like this is a new config option that was added in the last release, xref https://github.com/dask/dask/pull/11262 -- I wasn't sure if this was an existing config that got renamed and we're now undoing that rename.

I'm fine either with a deprecation or not given there won't be much "existing code" that uses this option. For future reference, adding config deprecations just means adding an entry to this dictionary 

https://github.com/dask/dask/blob/058f1b69ee3115d63046acf60e537d358462a466/dask/config.py#L703-L724

So it's relatively lightweight

**Review by jrbourbeau — APPROVED** [review]

**Comment by phofl** [comment]
Ah then I'll add the deprecation as well

**Comment by phofl** [comment]
It's a new option that is now being renamed, it didn't#t exist before the latest release

File: dask/dask.yaml [hunk 1]
```diff
@@ -21,11 +21,10 @@ dataframe:
 array:
   backend: "numpy"  # Backend array library for input IO and data creation
   chunk-size: "128MiB"
+  chunksize-tolerance: 1.25  # Tolerance for different algorithms when creating output chunks.
```
Comment (hendrikmakait) [inline dask/dask.yaml:24]: nit: Should we keep the spelling consistent with the option above?
```suggestion
  chunk-size-tolerance: 1.25  # Tolerance for different algorithms when creating output chunks.
```

**Review by hendrikmakait — APPROVED** [review]
Overall LGTM, one (non-blocking) question about spelling consistency.

**Pull request merged**